packages/pipeline/src/data_sources/ohlcv_external/crypto_compare.ts

```diff
@@ -0,0 +1,88 @@
+// tslint:disable:no-duplicate-imports
+import { AxiosResponse } from 'axios';
+import axios from 'axios';
+import promiseLimit = require('p-limit');
+
+import { merge, unfold } from 'ramda';
+
+import { TradingPair } from '../../utils/get_ohlcv_trading_pairs';
+
+export interface CryptoCompareOHLCVResponse {
+    Data: CryptoCompareOHLCVRecord[];
+}
+
+export interface CryptoCompareOHLCVRecord {
+    time: number; // in seconds, not milliseconds
+    close: number;
+    high: number;
+    low: number;
+    open: number;
+    volumefrom: number;
+    volumeto: number;
+}
+
+export interface CryptoCompareOHLCVParams {
+    fsym: string;
+    tsym: string;
+    e?: string;
+    aggregate?: string;
+    aggregatePredictableTimePeriods?: boolean;
+    limit?: number;
+    toTs?: number;
+}
+
+// tslint:disable:custom-no-magic-numbers
+const ONE_WEEK = 7 * 24 * 60 * 60 * 1000;
+const ONE_SECOND = 1000;
+
+const HISTORICAL_HOURLY_OHLCV_URI = 'https://min-api.cryptocompare.com/data/histohour';
+
+export class CryptoCompareOHLCVSource {
+    public readonly interval = ONE_WEEK; // the hourly API returns data for one week at a time
+    public readonly default_exchange = 'CCCAGG';
+    private readonly _url: string = HISTORICAL_HOURLY_OHLCV_URI;
```

Comment:
Is there a reason to set `_url` as an instance variable when it is already defined at the top of the file?

Reply:
I was thinking of parameterising the url, but won't this time.